core/src/main/java/org/apache/druid/guice/DruidSecondaryModule.java

```diff
@@ -36,6 +36,7 @@
 import javax.validation.Validator;
 import java.util.Properties;
 
+@LazySingleton
```

Comment:
So...  thinking about this, I realize that we *could* use such an annotation to require explicit bindings.  We could have an `@RequireExplicitBinding` scope annotation that would effectively throw an exception if it actually gets called.  This could be useful for any object that we expect to be explicitly bound.  If we did that, then the tests would need to have a way to explicitly bind that they expect this to be instantiated, maybe that's a good path?